agent/fw_drupal8.c

```diff
@@ -646,25 +655,28 @@ static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
 
   ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_implementation_map), hook_key,
                                 hook_val) {
-    if (nr_is_invalid_key_val_arr(hook_key, hook_val)) {
-      nrl_warning(NRL_FRAMEWORK,
-                  "hookImplementationsMap[hook]: invalid key or value");
+    if (nr_is_invalid_key_val_arr(hook_key, hook_val, "hook")) {
       return false;
     }
 
     ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(hook_val), class_key, class_val) {
-      if (nr_is_invalid_key_val_arr(class_key, class_val)) {
-        nrl_warning(NRL_FRAMEWORK,
-                    "hookImplementationsMap[class]: invalid key or value");
+      if (nr_is_invalid_key_val_arr(class_key, class_val, "class")) {
         return false;
       }
 
       ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(class_val), method_key,
                                     module_val) {
         if (NULL == method_key
             || 0 == nr_php_is_zval_valid_string(module_val)) {
-          nrl_warning(NRL_FRAMEWORK,
-                      "hookImplementationsMap[method]: invalid key or value");
+          if (NULL != method_key) {
```

Comment:
Could similarly split this to check 
if (NULL == method_key
then output the invalid key msg and return false
then check 0 == nr_php_is_zval_valid_string(module_val))
and output the invalid val msg and return false

Would make it more streamlined and save us the extra null check.

Reply:
b3531f513dc2811e0bd42420e75683261bd175f2